### PR TITLE
Add a SourceSegyFile, return that from SearchFiles

### PR DESCRIPTION
**Pull request by skairunner**
Added CloudStorageFile that incorporates some of the comments @amorken left on the V1 Files PR.

This work is related to [SEIS-417](https://cognitedata.atlassian.net/browse/SEIS-417). 

The plan is to implement this chunk first to enable SEIS-417.

### CODE REVIEW

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 1]
```diff
@@ -38,6 +39,34 @@ message OptionalMap {
     map<string, string> data = 1;
 }
 
+// File or dataset or cube derived from a single SEG-Y file
+message CloudStorageFile {
```
Comment (audunska) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:43]: I have to say I'm not thrilled about this name. The important thing about `File`s is not that they are stored in cloud storage, is it?

What about something like `SourceFile`, or `IngestionFile`, or maybe even `SegyFile`? Something to make it clear that these objects are mostly for ingestion, and it's the `SeismicStore` that's the central data store for querying.

One of the Hard Problems in computer science...

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 2]
```diff
@@ -38,6 +39,34 @@ message OptionalMap {
...
+    string uuid = 1;
+    int64 id = 2;
+    ExternalId external_id = 3;
+    string name = 4;
+    map<string, string> metadata = 5;
+    HeaderOffsets header_offsets = 6;
+}
+
+// Metadata related to interpreting SEG-Y files.
+message HeaderOffsets {
```
Comment (audunska) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:53]: To be pedandic, the `source_group_scalar_override` is not a header offset. What about `SegyOverrides`?

**Review by audunska — COMMENTED** [review]
API design is hard.

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto (hunk 1 above)
Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto:43]: i think segyfile or sourcefile is a good idea

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto (hunk 2 above)
Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto:53]: 👍

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto (hunk 1 above)
Comment (amorken) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto:43]: SourceSegYFile then? =)

File: cognite/seismic/protos/v1/seismic_service.proto [hunk 3]
```diff
@@ -134,5 +134,5 @@ service SeismicAPI {
      * Returns a stream of file objects, terminating all files matching the search criteria have been
      * returned.
      */
-    rpc SearchFiles(SearchFilesRequest) returns (stream File) {}
+    rpc SearchFiles(SearchFilesRequest) returns (stream SourceSegyFile) {}
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service.proto:137]: Should we rename the search api too? SearchSourceFiles?

OTT, this looks good to me. =)

**Review by audunska — APPROVED** [review]
Looks good to me. I have no particular opinion on the rpc name.

**Comment by amorken** [comment]
Yeah, I don't have a strong opinion on the RPC name either. So, again, looks good. Ponder the RPC name a bit and merge when ready. =)

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto (hunk 1 above)
Comment (skairunner) [reply cognite/seismic/protos/v1/seismic_service_datatypes.proto]: 👍

**Pull request merged**